scripts/build_project.sh

```diff
@@ -0,0 +1,39 @@
+#!/bin/bash
+
+# Author: Teghveer Singh Ateliey
+# Date: 2024-10-30
+
+# Check args
+if [ "$#" -ne 4 ]; then
+    echo "Usage: $0 <commit-hash> <repo-path> <project> <platform>"
+    exit
+fi
+
+# Assign inputs to variables
+COMMIT_HASH=$1
+REPO_PATH=$2
+PROJECT=$3
+PLATFORM=$4
+
+# Checks if directory exists
+if [ ! -d "$REPO_PATH" ]; then
+    echo "Error: Repository path '$REPO_PATH' does not exist."
+    exit
```

Comment:
these exits should return an error code. 2 would be good since it will differentiate between regular command errors which return 1 and script errors